apis/generate_certificate.php

```diff
@@ -2,7 +2,7 @@
 
 session_start();
 include 'db.php';
-if (isset($_SESSION['logged_in']) && isset($_SESSION['admin_id']))
+if ((isset($_SESSION['logged_in']) && isset($_SESSION['admin_id']))||(isset($_SESSION['admin_id'])&&isset($_SESSION['name'])))
```

Comment:
I guess admin_id and name is not suitable check for this. You can have another script for this or u can rename that admin id to user id and do.

Reply:
User details won't be there in the table unless they register. And once the feedback form is filled, it'll check whether rating, suggestions, comments are empty or not. If they are not empty, then it goes to this page. Since certificate name contains ID and name, I thought it would be better to reuse code by updating session variables.

Reply:
The user won't be registered by themselves. The organiser has to upload the email and names of all the participants first and then the user has to give the feedback. And the checking of rating, suggestions, comments are fine. you can store the primary key ie the user id in session and check if that user id exists in the table. If it does then get the certificate else, redirect to the feedback page. If u still have any doubt, Give me a call.